src/applications/disability-benefits/2346/schemas/address-schema.js

```diff
@@ -241,9 +277,48 @@ export const addressUISchema = (
           },
         },
       },
+      province: {
+        'ui:title': 'State/Province/Region',
+        'ui:options': {
+          hideIf: (formData, index) => {
+            let countryNamePath = `${path}.country`;
+            if (typeof index === 'number') {
+              countryNamePath = insertArrayIndex(countryNamePath, index);
+            }
+            const livesOnMilitaryBase = get(livesOnMilitaryBasePath, formData);
+            if (isMilitaryBaseAddress && livesOnMilitaryBase) {
+              return true;
+            }
+            const countryName = get(countryNamePath, formData);
+            return countryName === USA.name || !countryName;
+          },
+        },
+        'ui:required': (formData, index) => {
+          if (
+            formData.selectedAddress === 'newAddress' &&
+            (formData.newAddress.country &&
+              formData.newAddress.country !== 'United States')
+          ) {
+            let countryNamePath = `${path}.country`;
+            if (typeof index === 'number') {
+              countryNamePath = insertArrayIndex(countryNamePath, index);
+            }
+            const livesOnMilitaryBase = get(livesOnMilitaryBasePath, formData);
+            if (isMilitaryBaseAddress && livesOnMilitaryBase) {
+              return true;
```

Comment:
There is a lot of repeated code in the above three blocks. It'd be more DRY to move some of that code out into a function.

Reply:
I could refactor out these lines:

``let countryNamePath = `${path}.country`;``
 ` if (typeof index === 'number') {countryNamePath = insertArrayIndex(countryNamePath, index);
    }  
const livesOnMilitaryBase = get(livesOnMilitaryBasePath, formData);`

Maybe we can jump on a call if you see other opportunities.